operator-common/src/main/java/io/strimzi/operator/cluster/model/Ca.java

```diff
@@ -275,95 +337,136 @@ public CertAndKey generateSignedCert(String commonName, String organization) thr
-            }
-        }
-
         caCertSecret = secretCertProvider.createSecret(namespace, caCertSecretName, certData, labels,
                 Collections.singletonMap(ANNO_STRIMZI_IO_CA_CERT_GENERATION, String.valueOf(caCertGeneration)), ownerRef);
+
         caKeySecret = secretCertProvider.createSecret(namespace, caKeySecretName, keyData, labels,
-                Collections.emptyMap(), ownerRef);
+                Collections.singletonMap(ANNO_STRIMZI_IO_CA_KEY_GENERATION, String.valueOf(caKeyGeneration)), ownerRef);
     }
 
-    private boolean shouldCreateOrRenew(X509Certificate currentCert, String namespace, String clusterName) {
+    private Subject nextCaSubject(int version) {
+        Subject result = new Subject();
+        // Key replacements does not work if both old and new CA certs have the same subject DN, so include the
+        // key generation in the DN so the certificates appear distinct during CA key replacement.
+        result.setCommonName(commonName + " v" + version);
+        result.setOrganizationName(IO_STRIMZI);
+        return result;
+    }
+
+    private RenewalType shouldCreateOrRenew(X509Certificate currentCert, String namespace, String clusterName) {
         String reason = null;
-        boolean result = false;
-        if (this.caCertSecret == null
-                || this.caCertSecret.getData().get(CA_CRT) == null) {
-            reason = "CA certificate secret " + caCertSecretName + " is missing or lacking the key " + CA_CRT;
-            result = true;
-            this.caRenewed = this.caCertSecret != null;
-        } else if (caKeySecret == null
+        RenewalType renewalType = RenewalType.NOOP;
+        if (caKeySecret == null
                 || caKeySecret.getData().get(CA_KEY) == null) {
             reason = "CA key secret " + caKeySecretName + " is missing or lacking the key " + CA_KEY;
-            result = true;
-            this.caRenewed = caKeySecret != null;
+            renewalType = RenewalType.CREATE;
+        } else if (this.caCertSecret == null
+                || this.caCertSecret.getData().get(CA_CRT) == null) {
+            reason = "CA certificate secret " + caCertSecretName + " is missing or lacking the key " + CA_CRT;
+            renewalType = RenewalType.RENEW_CERT;
         } else if (this.caCertSecret.getMetadata() != null
                 && Annotations.booleanAnnotation(this.caCertSecret, ANNO_STRIMZI_IO_FORCE_RENEW, false)) {
             reason = "CA certificate secret " + caCertSecretName + " is annotated with " + ANNO_STRIMZI_IO_FORCE_RENEW;
-            result = true;
-            this.caRenewed = true;
+            renewalType = RenewalType.RENEW_CERT;
+        } else if (this.caKeySecret.getMetadata() != null
+                && Annotations.booleanAnnotation(this.caKeySecret, ANNO_STRIMZI_IO_FORCE_REPLACE, false)) {
+            reason = "CA key secret " + caKeySecretName + " is annotated with " + ANNO_STRIMZI_IO_FORCE_REPLACE;
```

Comment:
I think I answered this in the previous comment. I chose different annotations on purpose to avoid users accidentally applying the annotation to the wrong `Secret` and setting of a key replacement when they meant to do a cert renewal. I'm not sure `force-renew` is really the ideal name for the key case because a key doesn't expire, so "renewing" it is a little misleading.